### PR TITLE
http: api: implement vmm.ping

### PR DESCRIPTION
**Pull request by jcvenegas**
vmm.ping will help to check if http API server is up and
running.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

File: vmm/src/lib.rs [hunk 1]
```diff
@@ -299,6 +299,12 @@ impl Vmm {
         }
     }
 
+    fn vmm_info(&self) -> result::Result<VmmInfo, ApiError> {
+        Ok(VmmInfo {
+            version: env!("CARGO_PKG_VERSION").to_string(),
```
Comment (jcvenegas) [inline vmm/src/lib.rs]: Please note that the VERSION here is from the vmm create and not the one the Cloud hypervisor resulting binary, let me know should try to expose the cloud hypervisor binary create.

Comment (sameo) [reply vmm/src/lib.rs]: What's the difference between the two?

Comment (jcvenegas) [reply vmm/src/lib.rs]: today the vmm create is on version "0.1.0"

Comment (sameo) [reply vmm/src/lib.rs]: hmm, so it's taking the vmm crate package version...I would prefer if we use the main package version.
For that, I think we should pass it from main.rs to vmm as a `start_vmm_thread` parameter.

**Comment by sameo** [comment]
@jcvenegas Assuming we only care about this endpoint for checking if the VMM is alive and the HTTP API running, could we please instead:

* Remove the `vmm.info` endpoint for now until it's more clearly defined and needed.
* Add a new `vmm.ping` endpoint instead. It's more clearly defined than a `vmm.info` one.
* The ping endpoint would take `HEAD` or `GET` and return the VMM version (same response payload as the current `VmmInfo` one.

Does that make sense to you?

**Comment by jcvenegas** [comment]
@jcvenegas @rbradford make sense will update to use vmm.ping

**Comment by rbradford** [comment]
> * The ping endpoint would take `HEAD` or `GET` and return the VMM version (same response payload as the current `VmmInfo` one.

Maybe rename it `PingReply` as we might want a `VmmInfo` in the future

**Comment by sboeuf** [comment]
@jcvenegas what's the status on this? Did you have a chance to update the PR? It needs to be rebased btw.

**Comment by rbradford** [comment]
@jcvenegas Ping! :-) any update on this?

**Comment by rbradford** [comment]
We don't need two commits here. Please squash into one.

**Comment by rbradford** [comment]
Please disregard my last comment. I missed that vmminfo was already there but not implemented so having two commits seems reasonable.

File: vmm/src/api/openapi/cloud-hypervisor.yaml [hunk 2]
```diff
@@ -126,7 +126,7 @@ paths:
 components:
   schemas:
 
-    VmmInfo:
+    VmmPingResponse:
```
Comment (likebreath) [inline vmm/src/api/openapi/cloud-hypervisor.yaml]: @jcvenegas 
You seems to forget to replace the reference to `VmmInfo` with `VmmPingResponse` along with the change from `/vmm.info` to `/vmm.ping` : `$ref: '#/components/schemas/VmmInfo'` ->`$ref: '#/components/schemas/VmmPingResponse'` .

File: vmm/src/api/openapi/cloud-hypervisor.yaml [hunk 3]
```diff
@@ -12,16 +12,16 @@ servers:
 
 paths:
 
-  /vmm.info:
+  /vmm.ping:
     get:
-      summary: Returns general information about the cloud-hypervisor Virtual Machine Monitor (VMM).
+      summary: Requests a response to check if the VMM API server is available
```
Comment (sameo) [inline vmm/src/api/openapi/cloud-hypervisor.yaml]: Or simply: "Ping the VMM to check for API server availability"

**Pull request merged**